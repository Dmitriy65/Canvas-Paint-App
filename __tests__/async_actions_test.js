import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import "core-js/stable";
import "regenerator-runtime/runtime";
import {
  handleInputFile,
  HANDLE_FILE_ERROR
} from '../src/actions/fileActions';


const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);
describe('async actions tests', () => {
  let store;
  beforeEach(() => {
    store = mockStore({}); 
  });

  it('check for valid input file commands', async () => {
    const fileData =
      'C 20 4\n' +
      'L 22 2 22 6\n';
    const file = new File([fileData], 'input.txt', { type: 'text/plain' });

    await store.dispatch(handleInputFile(file));

    expect(store.getActions()[0]).toEqual({
      type: HANDLE_FILE_ERROR,
      payload: "Incorrect input commands.Please, try another commands!"
    });

  });

  it('check for existing data in input file', async () => {
    const file = new File([''], 'input.txt', { type: 'text/plain' });

    await store.dispatch(handleInputFile(file));

    expect(store.getActions()[0]).toEqual({
      type: HANDLE_FILE_ERROR,
      payload: "File doesn`t contains any data!"
    });

  });

  it('check for existing command to create canvas', async () => {
    const fileData =
      'L 10 2 6 2\n' +
      'R 16 1 20 3\n';

    const file = new File([fileData], 'input.txt', { type: 'text/plain' });

    await store.dispatch(handleInputFile(file));

    expect(store.getActions()[0]).toEqual({
      type: HANDLE_FILE_ERROR,
      payload: "Command to create canvas doesn`t exist!"
    });

  });

});