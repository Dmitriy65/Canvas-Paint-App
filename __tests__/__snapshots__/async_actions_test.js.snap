// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions creates async handleInputFile actions 1`] = `Array []`;
